validator/Dockerfile

```diff
@@ -0,0 +1,23 @@
+# syntax=docker/dockerfile:1
+FROM golang:1.22 AS build
+
+WORKDIR /app
+
+RUN go env -w GOMODCACHE=/root/.cache/go-mod
+RUN go env -w GOCACHE=/root/.cache/go-build
+
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . .
+
+RUN --mount=type=cache,target=/root/.cache/go-build,id=cache-go-build --mount=type=cache,target=/root/.cache/go-mod,id=cache-go-mod \
+    CGO_ENABLED=0 GOOS=linux go build -o /otel_config_validator
+
+FROM gcr.io/distroless/base-debian12
```

Comment:
A few reasons for that chose, one of which I know learn is false. The first one being that since I was inexperienced with Go I googled around to see what was the common base for a runner and it appeared to be this. That may be true.

Second, I thought it was smaller than alpine -- turns out its not. `static-debian12` i smaller (2MB) but `base-debian12` is actually 20MB. I can look into if static would be a good choice.

Last, I have just steered away from alpine for years now. It is my initial instinct to choose debian after seeing too many issues. That isn't to say alpine wouldn't be fine in this case, just the reason I personally didn't default to alpine. 

I'm happy with whatever, but would defer to what is the defacto standard (if any) for Go projects.

Reply:
> I'm happy with whatever, but would defer to what is the defacto standard (if any) for Go projects.

Agreed - I prefer following some standard vs inventing / justifying a new one.

The collector releases use alpine: https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol/Dockerfile#L1

Reply:
Oh, interesting. I guess we should follow their lead. Unless @MrAlias or @codeboten suggest otherwise to be the Go defactor standard?

Reply:
The collector uses alpine for building but then switches to `scratch` which is pretty standard for go tools

Reply:
@tsloughter do you want to switch to `scratch` and then merge this?

Reply:
@jack-berg yup, will do